File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack)):
            if haystack[i] == needle[0]:
                if i + len(needle) > len(haystack):
                    return -1
                check = haystack[i:i+len(needle)]
                if check == needle:
                    return i
        
        return -1

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    for i, v := range haystack {
        if v == rune(needle[0]) {
            if i + len(needle) > len(haystack) {
                return -1
            }
            
            candidate := haystack[i:i+len(needle)]
            if candidate == needle {
                return i
            }
        }
    }
    
    return -1
}